### PR TITLE
Add docker container with entrypoint for talker

### PR DESCRIPTION
**Pull request by vmayoral**
Changes include a new Dockerfile that creates a Docker container with the riot-ros2 code as part of an effort to merge this work and micro-ROS project efforts.

Container includes an entrypoint that launches automatically a native talker.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+RUN apt-get update
+RUN apt install -y gcc-arm-embedded
+RUN apt install -y wget
+
+#--------------------
+# Sources
+#--------------------
+WORKDIR /root
+RUN mkdir -p /root/ros2_riot_ws/src
+RUN cd /root/ros2_riot_ws && wget https://raw.githubusercontent.com/erlerobot/riot-ros2/base/riot-ros2.repos
```
Comment (astralien3000) [inline Dockerfile]: I would link to the main repository instead. Same for the other urls.

File: patches/sudo-tapsetup.patch [hunk 2]
```diff
@@ -0,0 +1,86 @@
+38,39c38,39
+<             sudo kldload if_bridge
```
Comment (astralien3000) [inline patches/sudo-tapsetup.patch:2]: I did not test it, but is removing sudo everywhere compulsory ?

**Review by astralien3000 — COMMENTED** [review]
Thanks for the contribution. I will test it as soon as possible.
I have just 2 comments/questions so far.

File: patches/sudo-tapsetup.patch (hunk 2 above)
Comment (vmayoral) [reply patches/sudo-tapsetup.patch:2]: @astralien3000 to avoid doing some Docker hacks yes, it is the simplest way to go.

File: Dockerfile (hunk 1 above)
Comment (vmayoral) [reply Dockerfile]: +1 for this. That's not a problem. I can probably do this tomorrow morning.

**Comment by astralien3000** [comment]
I pushed some change that addressed my comment on the url, and which also should fix #4.
Are you ok with that ?

**Comment by vmayoral** [comment]
Totally ok with the changes added. Thanks.

**Comment by astralien3000** [comment]
In that case, I think I will merge. Providing a Docker container can be interesting.

**Comment by vmayoral** [comment]
Fantastic! FYI, I'm putting together a prototype for NuttX at https://github.com/erlerobot/riot-ros2/tree/nuttx. I will probably just try replacing the RTOS and re-use the rest (including the ndn layer which i guess I'll need to port to NuttX as well).

Let's see how far i can get.

**Comment by astralien3000** [comment]
I think porting the ndn layer would be very hard (or even impossible). Indeed [this library](https://github.com/astralien3000/ndn-riot) makes an extensive use of RIOT-specific APIs, that seems not compatible with anything else I know.
I would advise you to try to use a "hollow" rmw, [like the one I started with](https://github.com/astralien3000/riot-ros2/blob/v0.2/rmw_fake/src/functions.cpp) to make sure everything work well before having a real rmw implementation. (the example I gave you is based on the beta2 API, which is not compatible with Arment, you may need a little of work to adapt).

**Comment by vmayoral** [comment]
Thanks, that's a good piece of feedback. I will do so then.

**Pull request merged**